public/app/features/explore/Logs/Logs.tsx

```diff
@@ -433,37 +436,26 @@ const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
     [props.eventBus]
   );
 
-  const onLogsContainerRef = useCallback(
-    (node: HTMLDivElement) => {
-      logsContainerRef.current = node;
-
-      // In theory this should be just a function passed down to LogRows but:
-      // - LogRow.componentDidMount which calls scrollIntoView is called BEFORE the logsContainerRef is set
-      // - the if check below if (logsContainerRef.current) was falsy and scrolling doesn't happen
-      // - and LogRow.scrollToLogRow marks the line as scrolled anyway (and won't perform scrolling when the ref is set)
-      // - see more details in https://github.com/facebook/react/issues/29897
-      // We can change it once LogRow is converted into a functional component
-      setScrollIntoView(() => (element: HTMLElement) => {
-        if (config.featureToggles.logsInfiniteScrolling) {
-          if (logsContainerRef.current) {
-            topLogsRef.current?.scrollIntoView();
-            logsContainerRef.current.scroll({
-              behavior: 'smooth',
-              top: logsContainerRef.current.scrollTop + element.getBoundingClientRect().top - window.innerHeight / 2,
-            });
-          }
-
-          return;
-        }
-        const scrollElement = props.scrollElement;
-
-        if (scrollElement) {
-          scrollElement.scroll({
+  const scrollIntoView = useCallback(
+    (element: HTMLElement) => {
+      if (config.featureToggles.logsInfiniteScrolling) {
+        if (logsContainerRef.current) {
+          topLogsRef.current?.scrollIntoView();
+          logsContainerRef.current.scroll({
             behavior: 'smooth',
-            top: scrollElement.scrollTop + element.getBoundingClientRect().top - window.innerHeight / 2,
+            top: logsContainerRef.current.scrollTop + element.getBoundingClientRect().top - window.innerHeight / 2,
           });
         }
-      });
+
+        return;
+      }
+
+      if (props.scrollElement) {
```

Comment:
optional nit: define scrollElement in props destructuring and remove the `props.`